moveit_plugins/moveit_ros_control_interface/src/controller_manager_plugin.cpp

```diff
@@ -236,6 +236,9 @@ class MoveItControllerManager : public moveit_controller_manager::MoveItControll
         getAbsName("controller_manager/list_controllers"));
     switch_controller_service_ = node_->create_client<controller_manager_msgs::srv::SwitchController>(
         getAbsName("controller_manager/switch_controller"));
+
+    std::unique_lock<std::mutex> lock(controllers_mutex_);
```

Comment:
```suggestion
    std::lock_guard<std::mutex> lock(controllers_mutex_);
```

I don't see you manually unlocking this or using it with a condition variable so you should be able to use a unique_lock.